includes/class-move.php

```diff
@@ -161,4 +161,74 @@ private static function update_blog_also_known_as( $from ) {
 
 		\update_option( 'activitypub_blog_user_also_known_as', $also_known_as );
 	}
+
+	/**
+	 * Change domain for all ActivityPub Actors.
+	 *
+	 * This method handles domain migration according to the ActivityPub Data Portability spec.
+	 * It stores the old domain and calls Move::internally for each available profile.
+	 * It also caches the JSON representation of the old Actor for future lookups.
+	 *
+	 * @param string $from The old domain.
+	 * @param string $to   The new domain.
+	 *
+	 * @return array Array of results from Move::internally calls.
+	 */
+	public static function change_domain( $from, $to ) {
+		// Get all actors that need to be migrated.
+		$actors = Actors::get_collection();
+
+		// Also include the blog actor if active.
+		if ( ! is_user_type_disabled( 'blog' ) ) {
+			$blog_actor = Actors::get_by_id( Actors::BLOG_USER_ID );
+			if ( ! \is_wp_error( $blog_actor ) ) {
+				$actors[] = $blog_actor;
+			}
+		}
+
+		$results = array();
+
+		// Process each actor.
+		foreach ( $actors as $actor ) {
+			$actor_id = $actor->get_id();
+
+			// Replace the old domain with the new domain in the actor ID.
+			$new_actor_id = str_replace( $from, $to, $actor_id );
+
+			// Call Move::internally for this actor.
+			$result = self::internally( $actor_id, $new_actor_id );
+
+			// Save the current actor data after migration.
+			self::cache_old_actor_data( $actor );
+
+			$results[] = array(
+				'actor'  => $actor_id,
+				'result' => $result,
+			);
+		}
+
+		// Store the old domain for future reference.
+		\update_option( 'activitypub_old_domain', $from );
+
+		return $results;
+	}
+
+	/**
+	 * Cache the old actor data for future lookups.
+	 *
+	 * @param object $actor The actor object.
+	 */
+	private static function cache_old_actor_data( $actor ) {
```

Comment:
Should we call it `save_old_actor_data()`?

Reply:
Sounds good to me